geminidr/core/parameters_preprocess.py

```diff
@@ -53,6 +53,20 @@ class dilateObjectMaskConfig(config.Config):
     dilation = config.RangeField("Dilation radius (pixels)", float, 1., min=0)
     repeat = config.Field("Allow dilation of already-dilated image?", bool, False)
 
+
+class fixPixelsConfig(config.Config):
+    suffix = config.Field("Filename suffix", dtype=str,
+                          default="_pixelsFixed", optional=True)
+    regions = config.Field("Regions to fix", dtype=str, optional=True)
+    regions_file = config.Field("Path to a file containing the regions to fix",
+                                dtype=str, optional=True)
+    axis = config.Field("Axis over which the interpolation is done",
```

Comment:
Shouldn't there be a default defined for `axis`?  None or zero for automatic mode, or something?

Reply:
The default is None but there is no need to specify it here, I think it's implicit with `optional=True`. 
`showpars` gives this:
```
 Name			Current setting

suffix               '_pixelsFixed'       Filename suffix
regions              None                 Regions to fix
regions_file         None                 Path to a file containing the regions to fix
axis                 None                 Axis over which the interpolation is done
use_local_median     False                Use a local median filter for single pixels?
```
0 would be the first axis so it cannot be the value for the automatic mode.

Reply:
I'm not a fan of the long Config doc string, because it's inconsistent with all the other strings. There's a large discussion here about what showpars should do compared to the more extensive docstrings in the primitives (and how to access those docstrings from the shell). My thought was always that showpars would behave like IRAF's lpar and provide a list of parameters with very short descriptions (just aides-memoires) whereas detailed help, more analogous to IRAF's help, (which would probably be the primitive's docstring) would appear somewhere else (I think I suggested smaug as the command, for SMArt User Guide). My main argument against having long strings in showpars is that the user almost certainly will want the long explanatory text from smaug if they're trying to learn the primitive for the first time and we obviously don't want to duplicate the text.

I also don't see the need for separate `regions` and `regions_file` parameters. Generally I like to keep the number of parameters small, perhaps to an unreasonable extent, but I think a single parameter works here.

Reply:
Overloaded input parameters are dangerous.  (re: using only regions for strings and file name)

Reply:
"Dangerous" how? I can't see any likely confusion between `my_regions.txt` and a string like `450,521;430:437,513:533`

Reply:
Messy in the code.

Reply:
At least in this specific case, the two inputs are strings.  That helps a bit, but it requires a fair bit of parsing and checking the nature of the string, "is this a file? or is this a region string?"

I'm the opposite of you, I don't like overloading input parameters.  I'm not against it, but I will always question it, to make sure that when we do it, we do it for the right reasons.

Reply:
I think it generally makes things easier for the user because they only have to remember the name of a single parameter and it's obvious to them whether it's a filename or a string (and it's not hard for the code to tell). Having parameters with similar names that do the same thing can lead to forgetting the names of parameters. If the code becomes messier, well that's secondary if it benefits the user experience. But having two parameters doesn't always make the code clean because sometimes you have to decide what to do if they're both set and should probably raise a warning (the code as written silently ignores the file, but I suggest that _both_ could be applied since they're not mutually exclusive and there might be advantages to this for the user).

Reply:
All I'm saying is think careful before just going for the knee-jerk reaction "I don't like input parameters, let's get rid of some."   

This is a case where I'm okay either way, but I push back to ensure it is a well thought decision, not just a reaction.

Reply:
The main reason to have `regions` and `regions_file` is to raise a better error if you pass it a path to a non existing file. With only one parameter I would to parse the invalid path and the user would get a misleading error (or I would change the error message to a generic one).

> you have to decide what to do if they're both set and should probably raise a warning (the code as written silently ignores the file, but I suggest that both could be applied since they're not mutually exclusive and there might be advantages to this for the user).

Yes indeed. Depending on what we choose I will improve this.

Reply:
> I'm not a fan of the long Config doc string, because it's inconsistent with all the other strings. There's a large discussion here about what showpars should do compared to the more extensive docstrings in the primitives (and how to access those docstrings from the shell).

Yes, I don't like either the long string here, but it was mostly to trigger the discussion. How users are expected to see the full docstring ? Maybe showpars could show it as well ?